### PR TITLE
Add release docs for PRs

### PR DESCRIPTION
**Pull request by normanrz**
For #535 and #531

### CODE REVIEW

**Comment by rabernat** [comment]
@normanrz can we also use this PR to bump the new version number from "unreleased" to 0.13.0?

**Comment by normanrz** [comment]
Sure. Is that something we have to do manually or is there some release automation?

**Comment by rabernat** [comment]
Here is the release process: https://numcodecs.readthedocs.io/en/stable/contributing.html#release-procedure

Before releasing, we need to manually update the release notes and also create a new space for "unreleased". So let's just use this PR to do all of that prep. Once it is merged, I will do the release.

File: docs/release.rst [hunk 1]
```diff
@@ -11,9 +11,11 @@ Release notes
 Unreleased
 ----------
 
-Maintenance
```
Comment (rabernat) [inline docs/release.rst:14]: I think we need new blank sections for Enhancements, Maintenance, etc. under unreleased.

Comment (normanrz) [reply docs/release.rst:14]: Got it.

**Pull request merged**